src/components/UserContext.js

```diff
@@ -0,0 +1,18 @@
+import { createContext, useState, useMemo } from 'react';
+const AuthContext = createContext();
+
+const AuthProvider = ({ children }) => {
+  let [user, setUser] = useState(null);
+  const providerValue = useMemo(() => ({ user, setUser }), [user, setUser]);
+
+  console.log(user);
+  const login = (email, pass) => {
+    setUser({ email: email, pass: pass });
+  };
+  const logout = (logout) => {
+    setUser(logout);
+  };
```

Comment:
No need to pass a value for logout, we should always set user to null when logging him out

```suggestion
  const logout = () => {
    setUser(null);
  };
```